pkg/ssh/ssh.go

```diff
@@ -153,19 +159,21 @@ func (s *Client) starting(ctx context.Context) error {
 			os.Exit(1)
 		}
 
-		level.Error(s.logger).Log("msg", "ssh client exited. restarting")
+		level.Info(s.logger).Log("msg", "ssh client exited. restarting", "exitCode", cmd.ProcessState.ExitCode())
 
 		// Check keys and cert validity before restart, create new cert if required.
 		// This covers the case where a certificate has become invalid since the last start.
 		// Do not return here: we want to keep trying to connect in case the PDC API
 		// is temporarily unavailable.
+		//
+		// They keymanager has logic to perform a background key refresh, but this
+		// logic should stay in place in case that is disabled.
 		if s.km != nil {
-			err := s.km.CreateKeys(ctx)
+			err := s.km.CreateKeys(ctx, false)
 			if err != nil {
 				level.Error(s.logger).Log("msg", "could not check or generate certificate", "error", err)
 			}
 		}
```

Comment:
I left this call to CreateKeys in, to keep the same behaviour as now if `CertCheckCertExpiryPeriod` is 0 (we want it to check & refresh whenever the connection is terminated). This will not create extra requests to the PDC API if the certificate is valid.